controller/users.go

```diff
@@ -155,6 +155,18 @@ func (c *UsersController) Create(ctx *app.CreateUsersContext) error {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
 	}
 
+	// we create a identity cluster relationship in cluster management service.
+	clusterURL := ctx.Payload.Data.Attributes.Cluster
+	err = c.app.ClusterService().AddIdentityToClusterLink(ctx.Context, identityID, clusterURL)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err":         err,
+			"identity_id": identityID,
+			"cluster_url": clusterURL,
+		}, "failed to link identity to cluster in cluster service")
+		// Not a blocker. Log the error and proceed.
```

Comment:
so what happens next if this error occurs? is there a way to retry the Identity/cluster link creation ?

Reply:
This error is critical enough to fail here instead of proceed I think.
We should roll back user creation here (hard delete user&identity from DB) and return an error, so, the reg app can repeat provisioning later.
But this is not that easy to implement and will complicate that PR. So, let's add a TODO here and create a new issue to address this in a separate PR.

Reply:
Created https://github.com/fabric8-services/fabric8-auth/issues/747